### ISSUE TITLE
Add hints to some exercises

### ISSUE THREAD

**Issue by abo64** [exercism/xscala#220]
Most exercises have their own favorable solutions and characteristic pitfalls.
To give people some guidance and avoid writing repetitive or similar code comments again and again
we should add a `HINTS.md` file to any exercise where it makes sense.

Especially the first couple of exercises should have such a file to assist beginners.
Another case is exercises dealing with special features of the language like `Option` or `Either`.

**Comment by ErikSchierboom** [comment]
Great idea. I would also suggest to include hints for the text-parsing exercises to Scala's parser combinator feature. 

For the `Option` and `Either` hints, we could link to the Neophyte's Guide to Scala, which has an excellent explanation of those topics.

**Comment by abo64** [comment]
@ErikSchierboom I am working on `Option` and `Either` hints, and I also love "Neophyte's Guide to Scala"! ;-)
Hope to have the PR ready soon.

@ErikSchierboom  @ricemery Btw, first I intended to add the whole list of exercises to this issue so that we can checkmark those containing hints, but decided against it because I thought we will have hints only for a couple but not all of them. Or do you disagree and we should add this list?

**Comment by ErikSchierboom** [comment]
I don't think we should do this for all exercises, but only for those where it makes sense. Also, in general, the smaller the PR the better, so my preference would be to create one PR per exercise.

Creating a list for all exercises that need hints might be wise, as it allows us to keep track of progress.

**Comment by abo64** [comment]
thx @ErikSchierboom 
I will add a such a list, and you guys feel free to add/modify it.
And let's not forget to keep it up to date.
